pennylane/workflow/qnode.py

```diff
@@ -1099,19 +1099,26 @@ def _execution_component(self, args: tuple, kwargs: dict, override_shots) -> qml
         full_transform_program.set_classical_component(self, args, kwargs)
         _prune_dynamic_transform(full_transform_program, inner_transform_program)
 
-        # pylint: disable=unexpected-keyword-arg
-        res = qml.execute(
-            (self._tape,),
-            device=self.device,
-            gradient_fn=self.gradient_fn,
-            interface=self.interface,
-            transform_program=full_transform_program,
-            inner_transform=inner_transform_program,
-            config=config,
-            gradient_kwargs=self.gradient_kwargs,
-            override_shots=override_shots,
-            **self.execute_kwargs,
-        )
+        with warnings.catch_warnings():
+            # TODO: remove this once the cycle for the arguements have finished, i.e. 0.39.
+            warnings.filterwarnings(
+                action="ignore",
+                message=r".*argument is deprecated and will be removed in version 0.39.*",
+                category=qml.PennyLaneDeprecationWarning,
+            )
+            # pylint: disable=unexpected-keyword-arg
+            res = qml.execute(
+                (self._tape,),
+                device=self.device,
+                gradient_fn=self.gradient_fn,
+                interface=self.interface,
+                transform_program=full_transform_program,
+                inner_transform=inner_transform_program,
+                config=config,
+                gradient_kwargs=self.gradient_kwargs,
+                override_shots=override_shots,
+                **self.execute_kwargs,

```

Comment:
Should we also deprecate the arguments to QNode that cause the call to `qml.execute` to have deprecated arguments? cc @trbromley

Reply:
There is a story for `max_expansion` in `QNode` that I didn't have time to work on: https://app.shortcut.com/xanaduai/story/64693/deprecate-max-expansion-kwarg-from-qnode-and-decompositions

And I believe `expand_fn` could be deprecated after this deprecation is done:
https://app.shortcut.com/xanaduai/story/67223/deprecate-expansion-strategy-in-qnode

@Shiro-Raven was working on that. There is no story for the other ones, afaik.